File: app/src/main/java/com/parabola/newtone/ui/dialog/DialogDismissLifecycleObserver.java
package com.parabola.newtone.ui.dialog;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.OnLifecycleEvent;

public final class DialogDismissLifecycleObserver implements LifecycleObserver {

    private AlertDialog dialog;

    public DialogDismissLifecycleObserver(@NonNull AlertDialog dialog) {
        this.dialog = dialog;
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    public void onDestroy(LifecycleOwner owner) {
        owner.getLifecycle().removeObserver(this);
        if (dialog != null) {
            dialog.dismiss();
            dialog = null;
        }
    }

}

File: app/src/main/java/com/parabola/newtone/presentation/base/DialogDismissLifecycleObserver.kt
package com.parabola.newtone.presentation.base

import androidx.appcompat.app.AlertDialog
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.OnLifecycleEvent

class DialogDismissLifecycleObserver(dialog: AlertDialog) : LifecycleObserver {

    private var dialog: AlertDialog?


    init {
        this.dialog = dialog
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    fun onDestroy(owner: LifecycleOwner) {
        owner.lifecycle.removeObserver(this)
        if (dialog != null) {
            dialog!!.dismiss()
            dialog = null
        }
    }

}
